domain-models-runtime/src/main/java/org/folio/rest/persist/SQLConnection.java

```diff
@@ -6,9 +6,20 @@
 public class SQLConnection {
 
   final PgConnection conn;
+  final int executionTimeLimit;
+  final long acquiringTime;
```

Comment:
You only need `final long timeoutTime`. Simply check whether `timeoutTime <= System.currentTimeMillis()`.